### PR TITLE
LPS-173580 & LPS-173488

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
@sandrodw3 @ruben-pulido Why not a StructsAction instead a MVC command?
IMHO, you are adding a MVC command in a portlet that is not related with the layout classed model usages.
If you create a StructsAction, you don't need a LayoutClassedModelUsagesHelper class since the logic should be in the same module and it can be an internal Util class.

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+				{Liferay.Language.get('there-are-no-usages')}
+			</p>
+		);
+	}
+
+	return (
+		<div className="cadmin">
+			<ClayList className="usages-list">
+				{loading ? (
+					<ClayLoadingIndicator className="my-7" size="sm" />
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:69]: Your 7 ? D:

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	const onUsagesLoad = useCallback(
+		(pageIndex) => {
+			setLoading(true);
+
+			const url = new URL(getUsagesURL);
+
+			url.searchParams.set(`${portletNamespace}pageIndex`, pageIndex);
+
+			fetch(url, {
+				method: 'GET',
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:40]: I believe the method 'GET' is the default one so we don't need to set it

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import '../css/ViewUsages.scss';
+
+export default function ViewUsages({getUsagesURL, portletNamespace}) {
+	const [activePage, setActivePage] = useState(1);
+	const [totalPages, setTotalPages] = useState(0);
+
+	const [items, setItems] = useState([]);
+	const [loading, setLoading] = useState(false);
+
+	const onUsagesLoad = useCallback(
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:31]: I believe we don't need to use a useCallback here, `getUsagesURL` and `portletNamespace`are constants, isn't it?

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+		<div className="cadmin">
+			<ClayList className="usages-list">
+				{loading ? (
+					<ClayLoadingIndicator className="my-7" size="sm" />
+				) : (
+					items.map(({id, name, type, url}) => (
+						<ClayList.Item flex key={id}>
+							<ClayList.ItemField className="p-0" expand>
+								<ClayList.ItemTitle className="mb-2 text-truncate">
+									<span title={name}>{name}</span>
```
Comment (victorg1991) [inline modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:75]: I assume this title right here is added to trigger the tooltip right? don't we need the lfr-portal-tooltip to trigger the clay tooltip?

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js (hunk 1 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:69]: 😂

**Comment by ruben-pulido** [comment]
> @sandrodw3 @ruben-pulido Why not a StructsAction instead a MVC command? IMHO, you are adding a MVC command in a portlet that is not related with the layout classed model usages. If you create a StructsAction, you don't need a LayoutClassedModelUsagesHelper class since the logic should be in the same module and it can be an internal Util class.

Thanks for the feedback @ealonso! It makes sense. I'll make that change

**Comment by ealonso** [comment]
@sandrodw3 
Why is necessary the new custom css?

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+								<ClayList.ItemField className="p-0">
+									<ClayButtonWithIcon
+										aria-label={Liferay.Language.get(
+											'view-usage'
+										)}
+										displayType="secondary"
+										onClick={() =>
+											openModal({
+												title: Liferay.Language.get(
+													'preview'
```
[the remark this answers is not in the document]
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:93]: It doesn't seem to be needed 🤷‍♂️I don't find any visual difference

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js (hunk 3 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:31]: Yeah, you are right, updated it!

File: modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js (hunk 4 above)
Comment (sandrodw3) [reply modules/apps/layout/layout-taglib/src/main/resources/META-INF/resources/layout_classed_model_usages_view/js/ViewUsages.js:75]: It doesn't seem to be needed here, it shows the Clay one without it

**Comment by sandrodw3** [comment]
> @sandrodw3 @ruben-pulido Why not a StructsAction instead a MVC command? IMHO, you are adding a MVC command in a portlet that is not related with the layout classed model usages. If you create a StructsAction, you don't need a LayoutClassedModelUsagesHelper class since the logic should be in the same module and it can be an internal Util class.

@ealonso I sent another PR to @ruben-pulido applying Victor's suggestions and I also used ClayTable component instead of ClayList (I didn't find out that ClayTable fits better the first time), so the CSS is not needed anymore. Ruben will send it when he applies what you suggested

**Comment by jkappler** [comment]
Resent here: https://github.com/liferay-echo/liferay-portal/pull/10945

**Pull request closed without merging**